api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -350,4 +355,129 @@ private OrderSet orderSetBuilder(boolean orderSetRetired, boolean orderSetMember
+	}
+
+	/**
+	 * @see OrderSetService#getOrderSetAttributeByUuid(String)
+	 */
+	@Test
+	public void getOrderSetAttributeByUuid_shouldReturnNullIfNoOrderSetAttributeHasTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetService service = Context.getOrderSetService();
+		assertNull(service.getOrderSetAttributeByUuid("not-a-uuid"));
+	}
+
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByUuid(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByUuid_shouldReturnTheOrderSetAttributeTypeWithTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		assertEquals("Audit Date", Context.getOrderSetService().getOrderSetAttributeTypeByUuid(
+		    "8516cc50-6f9f-33e0-8414-001e648eb67e").getName());
+	}
+
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByUuid(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByUuid_shouldReturnNullIfNoOrderSetAttributeTypeExistsWithTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		assertNull(Context.getOrderSetService().getOrderSetAttributeTypeByUuid("not-a-uuid"));
+	}
+
+	/**
+	 * @see OrderSetService#purgeOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void purgeOrderSetAttributeType_shouldCompletelyRemoveAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(initialOrderSetAttributeTypesCount + 0,
+				Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+		Context.getOrderSetService().purgeOrderSetAttributeType(Context.getOrderSetService().getOrderSetAttributeType(2));
+		assertEquals(1, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#retireOrderSetAttributeType(OrderSetAttributeType,String)
+	 */
+	@Test
+	public void retireOrderSetAttributeType_shouldRetireAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetAttributeType vat = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertFalse(vat.getRetired());
+		assertNull(vat.getRetiredBy());
+		assertNull(vat.getDateRetired());
+		assertNull(vat.getRetireReason());
+		Context.getOrderSetService().retireOrderSetAttributeType(vat, "for testing");
+		vat = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertTrue(vat.getRetired());
+		assertNotNull(vat.getRetiredBy());
+		assertNotNull(vat.getDateRetired());
+		assertEquals("for testing", vat.getRetireReason());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldCreateANewOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(initialOrderSetAttributeTypesCount + 0,
```

Comment:
What is the use of this assertion?

Reply:
its a sanity check

Reply:
A sanity check like assertEquals( x + 0, x)?

Reply:
@dkayiwa when i increment with `( x + 1, x)` it goes against the real order set attributes that i have in the dataset hence throwing an assertion failure between whats actual and whats is expected

Reply:
Let me ask differently, what value does `+ 0` add?

Reply:
honestly nothing and have removed it